index.js

```diff
@@ -94,3 +94,78 @@ exports.caps = {
+        case "datepicker": return "android.widget.DatePicker";
+        case "htmlview": return "android.widget.TextView";
+        case "image": return "org.nativescript.widgets.ImageView";
+        case "label": return "android.widget.TextView";
+        case "absolutelayout": return "android.view.View";
+        case "docklayout": return "android.view.View";
+        case "gridlayout": return "android.view.View";
+        case "stacklayout": return "android.view.View";
+        case "wraplayout": return "android.view.View";
+        case "listpicker": return "android.widget.NumberPicker";
+        case "listview": return "android.widget.ListView";
+        case "progress": return "android.widget.ProgressBar";
+        case "scrollview": return "android.view.View";
+        case "hscrollview": return "org.nativescript.widgets.HorizontalScrollView";
+        case "vscrollview": return "org.nativescript.widgets.VerticalScrollView";
+        case "searchbar": return "android.widget.SearchView";
+        case "segmentedbar": return "android.widget.TabHost";
+        case "slider": return "android.widget.SeekBar";
+        case "switch": return "android.widget.Switch";
+        case "tabview": return "android.support.v4.view.ViewPager";
+        case "textview":
+        case "textfield": return "android.widget.EditText";
+        case "timepicker": return "android.widget.TimePicker";
+        case "webview": return "android.webkit.WebView";
+    }
+
+    throw new Error("This "+name+" does not appear to tbe a standard NativeScript UI component");
+}
+
+function xpathiOS(name) {
+    switch (name) {
+        case "activityindicator": return "UIActivityIndicatorView";
+        case "button": return "UIButton";
+        case "datepicker": return "UIDatePicker";
+        case "htmlview": return "UITextView";
+        case "image": return "UIImageView";
+        case "label": return "TNSLabel";
+        case "absolutelayout": return "UIView";
+        case "docklayout": return "UIView";
+        case "gridlayout": return "UIView";
+        case "stacklayout": return "UIView";
+        case "wraplayout": return "UIView";
+        case "listpicker": return "UIPickerView";
+        case "listview": return "UITableView";
+        case "progress": return "UIProgressView";
+        case "scrollview":
+        case "hscrollview":
+        case "vscrollview": return "UIScrollView";
+        case "searchbar": return "UISearchBar";
+        case "segmentedbar": return "UISegmentedControl";
+        case "slider": return "UISlider";
+        case "switch": return "UISwitch";
+        case "tabview": return "UITabBarItem";
+        case "textview": return "UITextView"
+        case "textfield": return "UITextField";
+        case "timepicker": return "UIDatePicker";
+        case "webview": return "UIWebView";
+    }
+
+    throw new Error("This "+name+" does not appear to tbe a standard NativeScript UI component");
```

Comment:
I assume you are wanting then to be like this, where the returns all line up?:

    case "x":     // here
    case "yyyy":  // blah
    case "nn":    // blah